pkg/admission/validator/lakom/shoot.go

```diff
@@ -73,9 +97,16 @@ func (s *shoot) Validate(_ context.Context, new, _ client.Object) error {
 	if err := runtime.DecodeInto(s.decoder, lakomExt.ProviderConfig.Raw, lakomConfig); err != nil {
 		return fmt.Errorf("failed to decode providerConfig: %w", err)
 	}
-	if lakomConfig.Scope == nil {
-		return nil
+
+	errList := field.ErrorList{}
+
+	if lakomConfig.Scope != nil {
+		errList = append(errList, s.validateScopeType(providerConfigPath.Child("scope"), *lakomConfig.Scope)...)
+	}
+	if lakomConfig.PublicKeysSecretReference != nil {
+		// TODO: Can the secret be checked?
```

Comment:
Could you elaborate on this?
You're saying that admission controller will have to fetch all secrets on the virtual cluster and find the one that corresponds to the resource?

Reply:
I see what you mean. But it turns out my question was simpler than that.
When reconciling the extension resource, we fetch the secret that is in the same namespace as the extension.

When validating the shoot on the runtime cluster, I thought there wasn't going to be a way to easily find the secret, since it's in the project namespace. Turns out the project namespace is the shoot namespace that we already have, so there's no issue here. It's just a simple get via the client.

I'll implement this the same way they do it in the registry cache: https://github.com/gardener/gardener-extension-registry-cache/blob/058fb0be0e353ab5c7a97a0b48be2c533da101e4/pkg/admission/validator/cache/shoot.go#L114-L135

Reply:
It was confusing to me that when reconciling the extension resource, we search for the secret from the shoot namespace on the seed, while when validating the shoot, we fetch the secret from the project namespace on the virtual cluster.